torch/distributed/fsdp/_optim_utils.py

```diff
@@ -1161,7 +1141,63 @@ def _map_param_id_to_optim_keys(
                 )
         raise RuntimeError(error_msg)
 
-    return r0_param_id_to_optim_state_key, optim_state_key_to_param_id
+
+def _map_param_id_to_optim_keys(
+    optim_state_dict: Dict[str, Any],
+    group: Optional[dist.ProcessGroup],
+    param_id_to_param: List[nn.Parameter],
+    param_to_fqns: Dict[nn.Parameter, List[str]],
+    fqn_to_fsdp_param_info: Dict[str, FSDPParamInfo],
+    merge_key: bool = False,
```

Comment:
Do we need `merge_key=True` for `use_orig_params=True` to handle the fact that ranks who do not have any part of an original parameter will not have any optimizer state for that parameter?

Then, we also have `merge_key=False` to preserve the existing behavior for `use_orig_params=False`?

Reply:
That's correct.